scalableminds/webknossos
# Add option to expand and collapse all children of segment group

### Dev instance
https://collapseallsegmentgroupchildren.webknossos.xyz/

### Steps to test:
- go to segments tab in any annotation
- click some segments and create some groups, groups should be nested for at least three levels
- right click any group to access group context menu
- if the segment group has any subgroups, there should be one or two new items: "expand/collapse all subgroups".
- if the group is expanded and some subgroups are expanded and some are collapsed, both entries should be shown
- if the group itself is expanded and all subgroups are collapsed, you should only be able to expand them. the opposite is true if all subgroups are expanded.
- if the group itself is collapsed, you should only be able to expand all subgroups. the group itself will be expanded aswell.
~~- # 7827: Focus in segment list should also work for segments in a collapsed group~~

### TODOs:

- [x] switch label
- [x] default: all expanded
- [x] recursively expand subgroups

### Issues:
- fixes #7829 

------
(Please delete unneeded items, merge only when none are left open)
